Fix errors caught by pylint.

pylint src
************* Module src.notification_roles_bot
src/notification_roles_bot.py:7:0: C0301: Line too long (211/100) (line-too-long)
src/notification_roles_bot.py:23:0: C0301: Line too long (232/100) (line-too-long)
src/notification_roles_bot.py:47:0: C0301: Line too long (103/100) (line-too-long)
src/notification_roles_bot.py:91:0: C0301: Line too long (106/100) (line-too-long)
src/notification_roles_bot.py:100:0: C0301: Line too long (104/100) (line-too-long)
src/notification_roles_bot.py:120:0: C0301: Line too long (104/100) (line-too-long)
src/notification_roles_bot.py:131:0: C0301: Line too long (101/100) (line-too-long)
src/notification_roles_bot.py:84:0: W0622: Redefining built-in 'list' (redefined-builtin)
src/notification_roles_bot.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src/notification_roles_bot.py:29:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:33:4: W0702: No exception type(s) specified (bare-except)
src/notification_roles_bot.py:38:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:39:11: C0121: Comparison 'role.mentionable == True' should be 'role.mentionable is True' if checking for the singleton value True, or 'bool(role.mentionable)' if testing for truthiness (singleton-comparison)
src/notification_roles_bot.py:44:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:55:0: C0103: Function name "nr" doesn't conform to snake_case naming style (invalid-name)
src/notification_roles_bot.py:55:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:63:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:71:8: W0101: Unreachable code (unreachable)
src/notification_roles_bot.py:84:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:88:8: W0101: Unreachable code (unreachable)
src/notification_roles_bot.py:97:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:117:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:125:8: W0101: Unreachable code (unreachable)
src/notification_roles_bot.py:140:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:155:0: C0116: Missing function or method docstring (missing-function-docstring)
src/notification_roles_bot.py:8:0: C0411: standard import "from typing import List, Optional" should be placed before "from discord import Embed, Guild, Member, Permissions, Role" (wrong-import-order)

--- a/src/notification_roles_bot.py
+++ b/src/notification_roles_bot.py
@@ -1,13 +1,21 @@
 #!/usr/bin/env python
 
-# Imports
+"""
+Notification Roles Bot is a Discord bot that simplifies the use of no permission, mentionable roles
+that a user wants to subscribe to.
+"""
+
 import os
 import sys
-from discord import Embed, Guild, Member, Permissions, Role
-from discord.ext.commands import Bot, bot_has_guild_permissions, CommandError, CommandInvokeError, CommandNotFound, Context, guild_only, MissingRequiredArgument, NoPrivateMessage, RoleConverter, TooManyArguments
 from typing import List, Optional
 
-# Strings
+from discord import Embed, Guild, Member, Permissions, Role
+from discord.ext.commands import (Bot, CommandError, CommandInvokeError,
+                                  CommandNotFound, Context,
+                                  MissingRequiredArgument, NoPrivateMessage,
+                                  RoleConverter, TooManyArguments,
+                                  bot_has_guild_permissions, guild_only)
+
 GUILD_CONTEXT_REQUIRED_ERROR   = 'Uh-oh...this command is only valid in a guild context!'
 ROLE_ADD_TO_GUILD_ERROR        = 'Uh-oh...the role "{0}" could not be added to your guild!'
 ROLE_ADD_TO_MEMBER_ERROR       = 'Uh-oh...the role {0.mention} could not be added to you!'
@@ -20,142 +28,159 @@ ROLE_NOT_FOUND_IN_GUILD_ERROR  = 'Uh-oh...your guild does not have the role "{0}
 ROLE_NOT_FOUND_IN_MEMBER_ERROR = 'Uh-oh...you don\'t have the role {0.mention}!'
 ROLE_REMOVED_FROM_MEMBER       = 'The role {0.mention} has been removed from you!'
 SYNTAX                         = 'Syntax: !nr {list | {add | sub[scribe] | unsub[scribe]} <role>}'
-UNHANDLED_EXCEPTION            = 'It looks like you found a bug in Notification Roles Bot. If you would like to help us out, please file an issue on [GitHub](https://github.com/erickyeagle/notification-roles-bot/issues). Thank you!'
+TOKEN_NOT_SET_ERROR            = 'The environment variable "NOTIFICATION_ROLES_BOT_TOKEN" is not '\
+                                 'set!'
+UNHANDLED_EXCEPTION            = 'It looks like you found a bug in Notification Roles Bot. If you '\
+                                 'would like to help us out, please file an issue on [GitHub]'\
+                                 '(https://github.com/erickyeagle/notification-roles-bot/issues). '\
+                                 'Thank you!'
 
-# Initializes the bot.
 bot: Bot = Bot(case_insensitive = True, command_prefix = '!', help_command = None)
 
-# Converts a role name or mention string into a Role object.
 async def convert_str_to_role(context: Context, role_str: str) -> Optional[Role]:
+    """ Converts a role name or mention string into a Role object. """
     try:
         role: Optional[Role] = await RoleConverter().convert(context, role_str)
         return role
-    except:
+    except: # pylint: disable=bare-except
         return None
 
-# Returns whether the role is a notification role. A role is a notification role if 1) it is
-# mentionable, 2) it has no permissions, and 3) the role is in the bot's roles list.
 def is_notification_role(guild: Guild, role: Role) -> bool:
-    return role.mentionable == True               and \
+    """
+    Returns whether the role is a notification role. A role is a notification role if:
+        1) it is mentionable
+        2) it has no permissions
+        3) the role is in the bot's roles list
+    """
+    return role.mentionable                       and \
            role.permissions == Permissions.none() and \
            role in guild.me.roles
 
-# Loads the Discord bot token from the environment and starts the bot.
 def run() -> None:
+    """ Loads the Discord bot token from the environment and starts the bot. """
     notification_roles_bot_token: Optional[str] = os.environ.get('NOTIFICATION_ROLES_BOT_TOKEN')
     if not notification_roles_bot_token:
-        print('The environment variable "NOTIFICATION_ROLES_BOT_TOKEN" is not set!', file = sys.stderr)
+        print(TOKEN_NOT_SET_ERROR, file = sys.stderr)
         return
     bot.run(notification_roles_bot_token)
 
-# This is the initial bot command entrance. This bot uses command groups for specific behavior.
-@bot.group(case_insensitive = True)
+@bot.group(name = 'nr', case_insensitive = True)
 @bot_has_guild_permissions(manage_roles = True, read_messages = True, send_messages = True)
 @guild_only()
-async def nr(context: Context) -> None:
+async def _nr(context: Context) -> None:
+    """ Bot command group entry point. """
     if context.invoked_subcommand is None:
         raise CommandInvokeError
 
-# Adds a notification role to the guild.
-@nr.command(ignore_extra = False)
+@_nr.command(name = 'add', ignore_extra = False)
 @bot_has_guild_permissions(manage_roles = True, read_messages = True, send_messages = True)
 @guild_only()
-async def add(context: Context, role_str: str) -> None:
+async def _add(context: Context, role_str: str) -> None:
+    """ Adds a notification role to the guild. """
     role: Optional[Role] = await convert_str_to_role(context, role_str)
     if role is not None:
-        await context.reply(embed = Embed(description = ROLE_FOUND_IN_GUILD_ERROR.format(role)))
+        embed: Embed = Embed(description = ROLE_FOUND_IN_GUILD_ERROR.format(role))
+        await context.reply(embed = embed)
         return
     guild: Optional[Guild] = context.guild
     if guild is None:
         raise NoPrivateMessage
-        return
     permissions: Permissions = Permissions().none()
     role = await guild.create_role(mentionable = True, name = role_str, permissions = permissions)
     if role is None:
-        await context.reply(embed = Embed(description = ROLE_ADD_TO_GUILD_ERROR.format(role_str)))
+        embed: Embed = Embed(description = ROLE_ADD_TO_GUILD_ERROR.format(role_str))
+        await context.reply(embed = embed)
         return
     await guild.me.add_roles(role)
-    await context.reply(embed = Embed(description = ROLE_ADDED_TO_GUILD.format(role)))
+    embed: Embed = Embed(description = ROLE_ADDED_TO_GUILD.format(role))
+    await context.reply(embed = embed)
 
-# Lists all notification roles for the current guild.
-@nr.command(ignore_extra = False)
+@_nr.command(name = 'list', ignore_extra = False)
 @bot_has_guild_permissions(manage_roles = True, read_messages = True, send_messages = True)
 @guild_only()
-async def list(context: Context) -> None:
+async def _list(context: Context) -> None:
+    """ Lists all notification roles for the current guild. """
     guild: Optional[Guild] = context.guild
     if guild is None:
         raise NoPrivateMessage
-        return
     roles: List[Role] = [role for role in guild.roles if is_notification_role(guild, role)]
     if roles is not None and len(roles) > 0:
-        await context.reply(embed = Embed(description = ", ".join(map(lambda role: role.mention, roles))))
+        embed: Embed = Embed(description = ", ".join(map(lambda role: role.mention, roles)))
+        await context.reply(embed = embed)
 
-# Subscribes a user to a notification role.
-@nr.command(aliases = ['sub'], ignore_extra = False)
+@_nr.command(name = 'subscribe', aliases = ['sub'], ignore_extra = False)
 @bot_has_guild_permissions(manage_roles = True, read_messages = True, send_messages = True)
 @guild_only()
-async def subscribe(context: Context, role_str: str) -> None:
+async def _subscribe(context: Context, role_str: str) -> None:
+    """ Subscribes a user to a notification role. """
     role: Optional[Role] = await convert_str_to_role(context, role_str)
     if role is None:
-        await context.reply(embed = Embed(description = ROLE_NOT_FOUND_IN_GUILD_ERROR.format(role_str)))
+        embed: Embed = Embed(description = ROLE_NOT_FOUND_IN_GUILD_ERROR.format(role_str))
+        await context.reply(embed = embed)
         return
     guild: Optional[Guild] = context.guild
     if not is_notification_role(guild, role):
-        await context.reply(embed = Embed(description = ROLE_NOT_COMPATIBLE_ERROR.format(role)))
+        embed: Embed = Embed(description = ROLE_NOT_COMPATIBLE_ERROR.format(role))
+        await context.reply(embed = embed)
         return
     member: Member = context.author
     if role in member.roles:
-        await context.reply(embed = Embed(description = ROLE_FOUND_IN_MEMBER_ERROR.format(role)))
+        embed: Embed = Embed(description = ROLE_FOUND_IN_MEMBER_ERROR.format(role))
+        await context.reply(embed = embed)
         return
     await member.add_roles(role)
-    await context.reply(embed = Embed(description = ROLE_ADDED_TO_MEMBER.format(role)))
+    embed: Embed = Embed(description = ROLE_ADDED_TO_MEMBER.format(role))
+    await context.reply(embed = embed)
 
-# Unsubscribes a user from a notification role.
-@nr.command(aliases = ['unsub'], ignore_extra = False)
+@_nr.command(name = 'unsubscribe', aliases = ['unsub'], ignore_extra = False)
 @bot_has_guild_permissions(manage_roles = True, read_messages = True, send_messages = True)
 @guild_only()
-async def unsubscribe(context: Context, role_str: str) -> None:
+async def _unsubscribe(context: Context, role_str: str) -> None:
+    """ Unsubscribes a user from a notification role. """
     role: Optional[Role] = await convert_str_to_role(context, role_str)
     if role is None:
-        await context.reply(embed = Embed(description = ROLE_NOT_FOUND_IN_GUILD_ERROR.format(role_str)))
+        embed: Embed = Embed(description = ROLE_NOT_FOUND_IN_GUILD_ERROR.format(role_str))
+        await context.reply(embed = embed)
         return
     guild: Optional[Guild] = context.guild
     if guild is None:
         raise NoPrivateMessage
-        return
     if not is_notification_role(guild, role):
-        await context.reply(embed = Embed(description = ROLE_NOT_COMPATIBLE_ERROR.format(role)))
+        embed: Embed = Embed(description = ROLE_NOT_COMPATIBLE_ERROR.format(role))
+        await context.reply(embed = embed)
         return
     member: Member = context.author
     if not role in member.roles:
-        await context.reply(embed = Embed(description = ROLE_NOT_FOUND_IN_MEMBER_ERROR.format(role)))
+        embed: Embed = Embed(description = ROLE_NOT_FOUND_IN_MEMBER_ERROR.format(role))
+        await context.reply(embed = embed)
         return
     await member.remove_roles(role)
-    await context.reply(embed = Embed(description = ROLE_REMOVED_FROM_MEMBER.format(role)))
+    embed: Embed = Embed(description = ROLE_REMOVED_FROM_MEMBER.format(role))
+    await context.reply(embed = embed)
     if len(role.members) == 1 and guild.me in role.members:
         await role.delete()
 
-# The default command error handler provided by the bot.
 @bot.event
 async def on_command_error(context: Context, error: CommandError) -> None:
+    """ The default command error handler provided by the bot. """
     if isinstance(error, CommandNotFound):
         return
     if isinstance(error, (CommandInvokeError, MissingRequiredArgument, TooManyArguments)):
-        await context.reply(embed = Embed(description = SYNTAX))
+        embed: Embed = Embed(description = SYNTAX)
+        await context.reply(embed = embed)
         return
     if isinstance(error, NoPrivateMessage):
-        await context.reply(embed = Embed(description = GUILD_CONTEXT_REQUIRED_ERROR))
+        embed: Embed = Embed(description = GUILD_CONTEXT_REQUIRED_ERROR)
+        await context.reply(embed = embed)
         return
     embed: Embed = Embed()
     embed.add_field(name = error, value = UNHANDLED_EXCEPTION)
     await context.reply(embed=embed)
 
-# Called when the client is done preparing the data received from Discord.
 @bot.event
 async def on_ready() -> None:
+    """ Called when the client is done preparing the data received from Discord. """
     print('\n{0.user} is online!'.format(bot))
 
-# The "main" method essentially. The body of this conditional is executed when this script is
-# executed as the main module (ie. not imported by another module)
 if __name__ == '__main__':
     run()

